core-contracts/PriceOracle/src/main/java/finance/omm/score/core/oracle/PriceOracleImpl.java

```diff
@@ -0,0 +1,138 @@
+package finance.omm.score.core.oracle;
+
+import finance.omm.core.score.interfaces.PriceOracle;
+import finance.omm.libs.address.AddressProvider;
+import finance.omm.libs.address.Contracts;
+import finance.omm.score.core.oracle.exception.PriceOracleException;
+import finance.omm.score.core.oracle.token.BaseToken;
+import finance.omm.score.core.oracle.token.sICXToken;
+import finance.omm.score.core.oracle.token.Token;
+import score.Address;
+import score.Context;
+import score.VarDB;
+import score.annotation.External;
+
+import java.math.BigInteger;
+import java.util.Map;
+
+import static finance.omm.utils.math.MathUtils.ICX;
+import static finance.omm.utils.math.MathUtils.exaMultiply;
+
+public class PriceOracleImpl extends AddressProvider implements PriceOracle {
+
+    private static final String TAG = "Price Oracle Proxy";
+
+    private static final String OMM_POOL = "ommPool";
+    private static final String BALN = "BALN";
+    private static final String OMM = "OMM";
+
+    private VarDB<String> ommPool = Context.newVarDB(OMM_POOL,String.class);
+
+    private final String[] STABLE_TOKENS = {"USDS","bnUSD"};
+    private static final Token[] TOKENS = new Token[] {
+            new BaseToken("USDS", "USDS"),
+            new sICXToken("sICX","ICX"),
+            new BaseToken("IUSDC","USDC")
+    };
+
+
+    public PriceOracleImpl(Address addressProvider) {
+        super(addressProvider, false);
+        if (ommPool.get() == null) {
+            ommPool.set(OMM);
+        }
+    }
+
+    @External(readonly = true)
+    public String name() {
+        return "Omm " + TAG;
+    }
+
+    @External
+    public void setOMMPool(String _value) {
+        checkOwner();
+        ommPool.set(_value);
+    }
+
+    @External(readonly = true)
+    public String getOMMPool() {
+        return ommPool.get();
+    }
+
+    @External(readonly = true)
+    public BigInteger get_reference_data(String _base, String _quote) {
+        if (_base.equals(OMM)){
+            return getOmmPrice(_quote);
+        }
+        else {
+            return getPrice(_base, _quote);
+        }
+    }
+
+    private BigInteger getPrice(String _base, String _quote){
+        for (String token : STABLE_TOKENS) {
+            if (token.equals(_base)) {
+                return ICX;
```

Comment:
instead of returning ICX,please introduce new variable called ONE_USD and return it